### DISCUSSION TITLE
DateRangePicker with one date only (start only or end only)

### DISCUSSION THREAD

**Discussion by djmnz** [adobe/react-spectrum#5843] [Q&A]
We would like to have the _DateRangePicker_ to raise the OnChange event whenever either the Start Date or End Date is filled in.

Currently the onChange event is raised however there is no date available on the event itself until both dates are selected.

Is there anyway to achieve this outcome?

Somewhat related to this issue: https://github.com/adobe/react-spectrum/discussions/4859

**Comment by devongovett** [comment]
You could use two separate DatePickers rather than a DateRangePicker?

**Reply by djmnz** [reply]
yeah that's the conclusion we were getting to. :)

We were hoping to some "easy" way to change/receive the onChangeEvent. The date range picker validation and behaviour is pretty good except that in this case we need an "incomplete" range.

Thanks for the answer
